launch_yaml/README.md

```diff
@@ -0,0 +1,103 @@
+# launch_yaml
+
+This package provides an abstraction of the YAML tree.
+
+## YAML front-end mapping rules
+
+### Accessing yaml attributes
+
+When having an YAML file like:
+
+```yaml
+tag:
+    attr1: '2'
+    attr2: 2
+    attr3: 2.0
+```
+
+If the entity `e` is wrapping `tag`, the following statement will be true:
+```python
+e.get_attr('attr1') == '2'
+e.get_attr('attr2', types='int') == 2
+e.get_attr('attr3', types='float') == 2.0
+```
+
+By default, `get_attr` returns an string and it does type checking. The following code will raise a `TypeError`:
+
+```python
+e.get_attr('attr1', types='int')
+e.get_attr('attr2', types='float')
+e.get_attr('attr3')
+```
+
+Allowed types are: 
+```python
+'str', 'int', 'float', 'bool', 'list[int]', 'list[float]', 'list[bool]', 'list[str]'
+```
+A combination of them can be specified with a tuple. e.g.: `('int', 'str')`.
+In that case, conversions are tried in order and the first successful conversion is returned.
+`types` can also be set to `guess`, which works in the same way as passing:
+
+```python
+'int', 'float', 'bool', 'list[int]', 'list[float]', 'list[bool]', 'list[str]', 'str'
+```
+
```

Comment:
@ivanpauno meta: IMHO, type deduction documentation is design document material.

Reply:
If you want, I can update https://github.com/ros2/design/pull/208 and https://github.com/ros2/design/pull/207, and create a new PR for yaml design doc.

Reply:
Eventually, yeah, but let's hold that for a minute. At least until we've settled here.